### PR TITLE
Fix brand concierge close event dataLayer edge case

### PR DESCRIPTION
**Pull request by ruhisingh1**
Please provide the Jira Issue your PR is for.

Jira ID:

Test URLs:

- Before: https://main--exlm--adobe-experience-league.aem.live

- After: https://bc-datalayer--exlm--adobe-experience-league.aem.live?martech=off

### CODE REVIEW

File: scripts/brand-concierge/brand-concierge.js [hunk 1]
```diff
@@ -418,7 +422,9 @@ function handleBrandConciergeClientEvent(event) {
   if (event.eventType === BC_EVENT_HISTORY_CLEARED) {
     bcConversationId = null;
     bcMessageNumber = 0;
-    bcHasMessage = false;
+    // bcHasMessage is intentionally left as-is: clearing the transcript starts a new
+    // conversation, but if a message was sent before the clear, a subsequent widget
+    // close should still be tracked as "close with message".
   }
```
[the remark this answers is not in the document]
Comment (ruhisingh1) [reply scripts/brand-concierge/brand-concierge.js:434]: Fixed

**Review by Ashwathsaha — APPROVED** [review]
